code/Style.md

```diff
@@ -70,6 +72,48 @@ func Example() (err error) {
 }
 ```
 
+Sometimes, there are situations where a function, calls another function that only needs to return the value or the error returned by the call; when the called function return errors not created or wrapped by `errs` package the error must be  wrapped or annotated, however it isn't needed to have a conditional to achieve such purpose, because the functions that `errs` package have ignore the `nil` value when they are passed through; for example
+
+```go
+func OpenDataFile(filename string) (*os.File, error) {
+	file, err := os.Open(filepath.Join("data", filename))
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	return file, nil
+}
+```
+
+should be written like
+
+```go
+func OpenDataFile(filename string) (*os.File, error) {
+	file, err := os.Open(filepath.Join("data", filename))
+	return file, errs.Wrap(err)
+}
+```
+
+### Conditionals for _Not Found_ error values
+
+It's pretty common in Go to find functions that return an error when a function should return a value but such value "isn't found". The most classic example of this is the [`sql.ErrNoRows`](https://golang.org/pkg/database/sql/#pkg-variables) which may be returned by function like [`sql#DB.QueryRow`](https://golang.org/pkg/database/sql/#DB.QueryRow). Under this circumstances we first check for such error and if it doesn't match then we do a second check for non `nil` error, rather than nesting them.
```

Comment:
This seems to be strictly in the realm of aesthetic preference. Does it really matter one way or the other?

Reply:
Not to me, I'm fine with both, it looked odd to me the one written here, but because that one is spread through the sources I wanted to figure it out if it's a convention for aligning my future contributions to such code style.

Reply:
The nesting thing is because of some of the code relies on checks `err == sql.ErrNoRows`... if we hide the error `sql.ErrNoRows`, then we lose the capability to check it.

However, `errs.Is` should help us fix these situations.

Reply:
This is more about the order of the conditionals.

Reply:
Ah, order wise, sometimes we've been using:

``` go
if err != nil {
    if err == sql.ErrNoRows {
        return nil, ErrUser.New("not found %s", username)
    }
    return nil, ErrUser.Wrap(err)
}
```

This is to make clear that when the error happens it's always terminated.

Reply:
That's what I use, what I've seen that's commonly used and to me it's more clear for the reason that you wrote, however I found the other order several times in the code and I thought that there was such convention or preference.

Reply:
Do we agree that we should use the most common way?

Reply:
I'm not yet sure either way, I haven't gone through the code-base to see what the motivations might be for the alternate approaches.